### PR TITLE
Interactive forms: render button widget annotations (checkboxes and radio buttons)

### PR DESCRIPTION
**Pull request by timvandermeij**
Replaces #7664.
Fixes a part of #7613.

The reference tests do not display selection marks in fields *only when checkbox/radio button styling is supported by the browser* because the rasterizer cannot handle that. This is not a problem for landing this patch, but we will add it to the tracking issue when this is merged.

### CODE REVIEW

**Review by yurydelendik — APPROVED** [review]
Looks good

File: src/core/annotation.js [hunk 1]
```diff
@@ -767,6 +769,79 @@ var TextWidgetAnnotation = (function TextWidgetAnnotationClosure() {
...
+                                                        renderForms) {
+      var operatorList = new OperatorList();
+
+      // Do not render form elements on the canvas when interactive forms are
+      // enabled. The display layer is responsible for rendering them instead.
+      if (renderForms) {
+        return Promise.resolve(operatorList);
+      }
+
+      if (this.appearance) {
```
Comment (yurydelendik) [inline src/core/annotation.js]: Add a TODO since we need to select AS based on value.

File: src/display/annotation_layer.js [hunk 2]
```diff
@@ -532,6 +541,94 @@ var TextWidgetAnnotationElement = (
...
+      var inputElement = document.createElement('input');
+      inputElement.disabled = this.data.readOnly;
+      inputElement.type = 'checkbox';
+      inputElement.id = this.data.fieldName;
+      if (this.data.fieldValue && this.data.fieldValue !== 'Off') {
+        inputElement.checked = true;
+      }
+      this.container.appendChild(inputElement);
+
+      var labelElement = document.createElement('label');
```
Comment (timvandermeij) [inline src/display/annotation_layer.js]: Done in the new commit.

File: src/core/annotation.js (hunk 1 above)
Comment (timvandermeij) [inline src/core/annotation.js]: This is already taken care of by the `getDefaultAppearance` function based on the `AS` dictionary entry. Refactoring this is already part of the tracking issue.

File: web/annotation_layer_builder.css [hunk 3]
```diff
@@ -97,6 +99,66 @@
   width: 115%;
 }
 
+.annotationLayer .buttonWidgetAnnotation.checkBox label,
+.annotationLayer .buttonWidgetAnnotation.radioButton label {
```
Comment (Snuffleupagus) [inline web/annotation_layer_builder.css]: Unless I'm mistaken, most of these rules are the same as those used above in https://github.com/mozilla/pdf.js/pull/7898/files#diff-6ebc03110e3e120aeeedb42bf7ba8d90R44, hence moving these lines there (with possible necessary overrides here) might make the code more readable/maintainable.

File: test/annotation_layer_test.css [hunk 4]
```diff
@@ -75,6 +77,26 @@
   padding-right: 0;
 }
 
+.annotationLayer .buttonWidgetAnnotation.checkBox label,
+.annotationLayer .buttonWidgetAnnotation.radioButton label {
```
Comment (Snuffleupagus) [inline test/annotation_layer_test.css]: Unless I'm mistaken, most of these rules are the same as those used above in https://github.com/mozilla/pdf.js/pull/7898/files#diff-7d2639c28c78de0283c24b2c5d56eb60R46, hence moving these lines there (with possible necessary overrides here) might make the code more readable/maintainable.

**Review by Snuffleupagus — COMMENTED** [review]
Given that https://bugzilla.mozilla.org/show_bug.cgi?id=418833 has now (finally) landed, most modern browsers should now support proper styling of Checkboxes and Radio buttons.
With that in mind: Would it be possible to use normal HTML elements, instead of the `label` hacks used here, since I believe that that would simplify the HTML, CSS and JavaScript a great deal here?

Yes this would mean that these elements wouldn't look as nice in browsers without support for styling them, but given the simplicity of that implementation I'd greatly prefer the approach with proper form elements!

File: test/annotation_layer_test.css (hunk 4 above)
Comment (timvandermeij) [inline test/annotation_layer_test.css]: Done in the new commit.

**Comment by timvandermeij** [comment]
Do you mind reviewing this PR again @Snuffleupagus, in particular the last commit, to check if it's better this way?

**Review by Snuffleupagus — APPROVED** [review]
*Since Yury has already reviewed most of this, I've mainly skimmed through the total diff.*

> [...] in particular the last commit, to check if it's better this way?

Yes, I think it's much better this way! By directly using the proper elements, the code becomes a lot cleaner and easier to follow!

I've got just one remaining question, and please keep in mind that "no" is a perfectly valid answer:
I've not looked too closely at the individual commits (besides the last one). but would it be possible/useful to fold some the smaller commits together before landing?

---
Also, when landing this, don't forget to add a note in the tracking issue about the flags that this PR doesn't implement; i.e. `NoToggleToOff` and `RadiosInUnison`, see http://www.adobe.com/content/dam/Adobe/en/devnet/acrobat/pdfs/PDF32000_2008.pdf#G11.2201104.

**Comment by timvandermeij** [comment]
Thank you. I managed to reduce the commits I added from five to three.

**Comment by timvandermeij** [comment]
I have updated the tracking issue with the remaining points.

**Pull request merged**